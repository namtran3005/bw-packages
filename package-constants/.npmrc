@bitwala-cryptobank-squad:registry=https://github.bitwa.la/_registry/npm/
