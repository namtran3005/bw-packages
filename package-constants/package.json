{
  "name": "@bitwala-cryptobank-squad/package-constants",
  "license": "SEE LICENSE IN LICENSE",
  "publishConfig": {
    "registry": "https://github.bitwa.la/_registry/npm"
  },
  "version": "1.3.2",
  "description": "Shared package for UI constants",
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "clean": "rm -rf build",
    "prepare": "yarn build",
    "test": "yarn test:lint && yarn test:coverage",
    "test:coverage": "yarn test:unit --coverage",
    "test:lint": "eslint 'src/**/*.{ts,tsx}' --color -c ./.eslintrc.js",
    "test:unit": "jest src/",
    "build": "yarn clean && tsc"
  },
  "jest": {
    "preset": "ts-jest",
    "testEnvironment": "node",
    "collectCoverageFrom": [
      "src/**/*{ts,tsx}"
    ],
    "coveragePathIgnorePatterns": [
      "/node_modules/",
      "/src/types/"
    ],
    "coverageReporters": [
      "text-summary"
    ],
    "coverageThreshold": {
      "global": {
        "statements": 100,
        "branches": 100,
        "functions": 58
      }
    }
  },
  "devDependencies": {
    "@bitwala/eslint-config": "0.4.2",
    "@types/jest": "26.0.15",
    "jest": "26.4.2",
    "ts-jest": "26.4.1",
    "typescript": "4.0.3"
  }
}
