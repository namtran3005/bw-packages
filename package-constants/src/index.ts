export * from './countries';
export * from './identificationDocuments';
export { germanStateCodes } from './germanStateCodes';
export { Locales, localeOptions } from './locales';
export { Industries } from './industries';
export { Positions, FreelancerPositions } from './positions';
export * from './referral';
export * from './countryDialingCodes';
export { extraTranslations } from './extraTranslations';
export * from './emails';
export { countryNames } from './countryNames';
export * from './cards';
export * from './communications';
export * from './trading';
export * from './wallets';
export * from './codes';
export * from './types';
