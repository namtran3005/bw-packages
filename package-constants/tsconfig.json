{
  "compilerOptions": {
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "outDir": "build",
    "rootDir": "src",
    "moduleResolution": "node",
    "target": "es5",
    "sourceMap": true,
    "declaration": true,
    "esModuleInterop": true,
    "isolatedModules": true
  },
  "exclude": [
    "node_modules",
    "build"
  ]
}
