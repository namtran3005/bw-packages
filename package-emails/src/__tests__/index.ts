import { Locales } from '@bitwala-cryptobank-squad/package-constants';
import { Emails, templates, TemplateNames } from '../';

const emails = new Emails(templates);

describe('Email class', () => {
  it('Should generate emails', () => {
    const html = emails.generateHTML(
      TemplateNames.MainResetPassword,
      Locales.en
    );
    expect(html).toMatchSnapshot();
  });
});
