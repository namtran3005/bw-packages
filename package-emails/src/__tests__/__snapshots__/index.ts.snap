// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email class Should generate emails 1`] = `
"
<mjml>
  <mj-head>
    <mj-style inline=\\"inline\\">a { text-decoration: none!important; color:#0038FF; }</mj-style>
    <mj-style inline=\\"inline\\"> .footer-link { text-decoration: underlined!important; color:#abadb7; }</mj-style>
    <mj-style>
      @font-face {
        font-family: \\"Signifier\\";
        src: url(\\"https://public.nuri.com/email-assets/fonts/signifier-web-extralight.woff2\\") format(\\"woff2\\"),
          url(\\"https://public.nuri.com/email-assets/fonts/signifier-web-extralight.woff\\") format(\\"woff\\");
        font-weight: 200;
        font-style: normal;
        font-display: block;
        unicode-range: U+000-5FF;
      }
      @font-face {
        font-family: \\"Inter\\";
        src: url(\\"https://public.nuri.com/email-assets/fonts/Inter-Black.ttf\\")
          format(\\"woff\\");
        font-weight: regular;
        font-style: normal;
        font-display: swap;
      }
      @font-face {
        font-family: \\"Inter\\";
        src: url(\\"https://public.nuri.com/email-assets/fonts/Inter-Bold.ttf\\")
          format(\\"woff\\");
        font-weight: bold;
        font-style: bold;
        font-display: swap;
      }
      @font-face {
        font-family: \\"Inter\\";
        src: url(\\"https://public.nuri.com/email-assets/fonts/Inter-Regular.ttf\\")
          format(\\"woff\\");
        font-weight: regular;
        font-style: normal;
        font-display: swap;
      }
      @font-face {
        font-family: \\"Inter\\";
        src: url(\\"https://public.nuri.com/email-assets/fonts/Inter-Medium.ttf\\")
          format(\\"woff\\");
        font-weight: 500;
        font-style: normal;
        font-display: swap;
      }
      @font-face {
        font-family: \\"Inter\\";
        src: url(\\"https://public.nuri.com/email-assets/fonts/Inter-SemiBold.ttf\\")
          format(\\"woff\\");
        font-weight: 600;
        font-style: normal;
        font-display: swap;
      }
    </mj-style>
    <mj-attributes>
      <mj-all font-family=\\"'Inter', sans-serif\\" align=\\"left\\" padding=\\"0px\\"/>
      <!-- Sections!-->
      <mj-class name=\\"content\\" border-radius=\\"9px\\" background-color=\\"#F0F0F0\\" padding-left=\\"6%\\" padding-right=\\"6%\\" padding-bottom=\\"48px\\" padding-top=\\"0px\\"></mj-class>
      <mj-class name=\\"content\\" background-color=\\"#F0F0F0\\" padding-left=\\"6%\\" padding-right=\\"6%\\" padding-bottom=\\"48px\\" padding-top=\\"0px\\"></mj-class>
      <mj-class name=\\"contact-support\\" padding-bottom=\\"32px\\" padding-top=\\"42px\\"></mj-class>
      <!-- Titles!-->
      <mj-class name=\\"h1\\" color=\\"#000000\\" padding=\\"10px 25px\\" font-size=\\"44px\\" font-weight=\\"800\\" letter-spacing=\\"-1.5px\\" line-height=\\"39px\\"/>
      <mj-class name=\\"h2\\" color=\\"#000000\\" font-family=\\"'Signifier', 'Times New Roman', serif\\" padding=\\"10px 25px 25px 25px\\" font-size=\\"32px\\" font-weight=\\"400\\" letter-spacing=\\"-1.5px\\" line-height=\\"36px\\" align=\\"left\\"/>
      <mj-class name=\\"h3\\" color=\\"#000000\\" padding=\\"10px 25px\\" font-size=\\"24px\\" font-weight=\\"800\\" letter-spacing=\\"-1px\\" line-height=\\"28px\\"/>
      <mj-class name=\\"h4\\" color=\\"#000000\\" padding=\\"8px 24px 0\\" font-size=\\"20px\\" font-weight=\\"800\\" letter-spacing=\\"-1px\\" line-height=\\"28px\\" />
      <!-- Text!-->
      <mj-text padding=\\"16px 25px\\" font-size=\\"18px\\" line-height=\\"24px\\" color=\\"#000000\\"></mj-text>
      <mj-font name=\\"Lato\\" href=\\"https://fonts.googleapis.com/css?family=Inter\\" />
      <mj-class name=\\"sub-text\\" font-size=\\"12px\\" line-height=\\"1.6\\" padding-top=\\"30px\\"/>
      <mj-class name=\\"support-help\\" font-size=\\"14px\\" line-height=\\"1.4\\" color=\\"#2C232E\\" weight=\\"bold\\" padding=\\"0\\"/>
      <mj-class name=\\"left\\" align=\\"left\\"/>
      <mj-class name=\\"bold\\" font-weight=\\"700\\"/>
      <mj-class name=\\"small-text\\" font-size=\\"12px\\"/>
      <mj-class name=\\"link-text\\" color=\\"#0038FF\\"/>
      <!-- Images!-->
      <mj-class name=\\"top-icon\\" width=\\"150px\\" height=\\"150px\\" alt=\\"Icon\\" padding-top=\\"24px\\" bottom-padding=\\"0\\" align=\\"center\\"/>
      <mj-class name=\\"support-face\\" width=\\"54px\\" height=\\"54px\\" alt=\\"Icon\\" padding-bottom=\\"10px\\"/>
      <mj-class name=\\"nuri-logo\\" width=\\"180px\\" height=\\"auto\\" alt=\\"Nuri\\" padding-top=\\"24px\\" bottom-padding=\\"0\\" align=\\"center\\" />
      <mj-class name=\\"content-image\\" width=\\"550px\\" height=\\"auto\\" alt=\\"Icon\\" padding=\\"16px 0px\\" bottom-padding=\\"0\\" align=\\"center\\" />
      <!-- Buttons!-->
      <mj-class name=\\"primary-button\\" align=\\"left\\" background-color=\\"#BEAAFF\\" color=\\"#2C232E\\" border-radius=\\"32px\\" font-size=\\"18px\\" font-weight=\\"500\\" line-height=\\"24px\\" padding=\\"10px\\" padding-left=\\"20px\\" border-radius=\\"0px\\"  inner-padding=\\"12px 45px 12px 45px\\"/>
      <mj-class name=\\"primary-button-preregister\\" background-color=\\"#00ADEE\\" color=\\"#FFFFFF\\" font-size=\\"18px\\" font-weight=\\"bold\\" line-height=\\"24px\\" padding=\\"10px 10px 48px 10px\\" border-radius=\\"0px\\"  inner-padding=\\"12px 45px 12px 45px\\"/>
      <mj-class name=\\"secondary-button\\" border=\\"2px solid #00ADEE;\\" background-color=\\"transparent\\" color=\\"#00ADEE\\" font-size=\\"18px\\" font-weight=\\"bold\\" line-height=\\"24px\\" padding=\\"10px\\" border-radius=\\"0px\\"  inner-padding=\\"12px 45px 12px 45px\\"/>
      <mj-social-element />
      <!-- All styles!-->
    </mj-attributes>
  </mj-head>


  <mj-body background-color=\\"#D4D2D2\\">
  
<mj-section background-color=\\"transparent\\" padding-top=\\"70px\\" padding-bottom=\\"49px\\">
<mj-column width=\\"50%\\">
  <mj-image mj-class=\\"nuri-logo\\" css-class=\\"nuri-logo\\" src=\\"https://public.nuri.com/email-assets/images/nuri-logo-primary-512x512.png\\" alt=\\"Nuri Logo\\" width=\\"100px\\"></mj-image>
</mj-column>
</mj-section>

    <mj-section mj-class=\\"content\\">
      <mj-column>
        <mj-image mj-class=\\"top-icon\\" src=\\"https://public.nuri.com/email-assets/images/main-reset_password-en.png\\" alt=\\"keys\\"></mj-image>
        <mj-text mj-class=\\"h2\\">Forgot your password?</mj-text>
        <mj-text>Hi {{firstName}},</mj-text>
        <mj-text>We received your request to reset your password. You can do so by clicking on the button below.</mj-text>
        <mj-button mj-class=\\"primary-button\\" href=\\"{{url}}\\">Reset my password</mj-button>
        <mj-text>Thanks,<br>Team Nuri</mj-text>
        <mj-text mj-class=\\"sub-text\\">If you didn't make this request, please contact customer support below.</mj-text>
      </mj-column>
    </mj-section>
    
<mj-section background-color=\\"transparent\\" padding=\\"0px\\" padding-bottom=\\"96px\\">
    <mj-column width=\\"75%\\">
    <mj-social icon-size=\\"30px\\" mode=\\"horizontal\\" padding-bottom=\\"32px\\" padding-top=\\"60px\\" align=\\"center\\">
    <mj-social-element name=\\"[twitter]-noshare\\" src=\\"https://public.nuri.com/email-assets/icons/twitter-icon.png\\" href=\\"https://twitter.com/nuribanking\\" padding=\\"5px 8px\\" background-color=\\"transparent\\" icon-size=\\"48px\\">
    </mj-social-element>
    <mj-social-element name=\\"[youtube]-noshare\\" src=\\"https://public.nuri.com/email-assets/icons/youtube-icon.png\\" href=\\"http://youtube.com/nuribanking\\" padding=\\"5px 8px\\" background-color=\\"transparent\\" icon-size=\\"48px\\">
    </mj-social-element>
    <mj-social-element name=\\"instagram\\" src=\\"https://public.nuri.com/email-assets/icons/instagram-icon.png\\" href=\\"https://www.instagram.com/NuriBanking/\\" padding=\\"5px 8px\\" background-color=\\"transparent\\" icon-size=\\"48px\\">
    </mj-social-element>
    <mj-social-element name=\\"telegram\\" src=\\"https://public.nuri.com/email-assets/icons/telegram-icon.png\\" href=\\"https://t.me/NuriBanking\\" padding=\\"5px 8px\\" background-color=\\"transparent\\" icon-size=\\"48px\\">
    </mj-social-element>
    </mj-social>
    <mj-text mj-class=\\"support-help\\" padding=\\"0\\" align=\\"center\\">
      You are getting this message because you signed up to Nuri.<br/><br/>
      Nuri GmbH<br/>
      Prinzessinnenstra√üe 19/20,<br/>
      10969 Berlin.<br/>
      Board of Management | Christoph Iwaniez, Jan Goslicki, Kristina Walcker-Mayer<br/>
      Commercial register | HRB173698B, Charlottenburg<br/>
        EU VAT ID | DE305362792
    </mj-text>
    </mj-column>
</mj-section>

  </mj-body>
</mjml>
"
`;
